src/allotropy/parsers/appbio_quantstudio_designandanalysis/structure/generic/creator.py

```diff
@@ -10,12 +11,28 @@
 
 
 class Creator(ABC):
-    EXPECTED_SHEETS: ClassVar[list[str]] = []
+    PLUGIN_REGEX: ClassVar[str]
+    EXPECTED_SHEETS: ClassVar[list[str]]
 
     @classmethod
-    def check_type(cls, reader: DesignQuantstudioReader) -> bool:
+    def check_sheets(cls, reader: DesignQuantstudioReader) -> bool:
         return all(reader.has_sheet(sheet_name) for sheet_name in cls.EXPECTED_SHEETS)
 
+    @classmethod
+    def check_plugin_name(cls, raw_plugin_name: str) -> bool:
+        return bool(search(cls.PLUGIN_REGEX, raw_plugin_name))
+
+    @classmethod
+    def check_experiment_type(
+        cls, reader: DesignQuantstudioReader, raw_plugin_name: str | None
+    ) -> bool:
+        if not cls.check_sheets(reader):
```

Comment:
```suggestion
        return cls.check_sheets(reader) and (raw_plugin_name is None or cls.check_plugin_name(raw_plugin_name))
```
You could, optionally, move `None` check for `raw_plugin_name` into `check_plugin_name`, and then it would just be `return cls.check_sheets(reader) and cls.check_plugin_name(raw_plugin_name)`